aten/src/ATen/cuda/CUDABlas.cpp

```diff
@@ -432,6 +431,36 @@ void ger(int64_t m, int64_t n, scalar_t alpha, scalar_t *x, int64_t incx, scalar
 template void ger<float>(int64_t m, int64_t n, float alpha, float *x, int64_t incx, float *y, int64_t incy, float *a, int64_t lda);
 template void ger<double>(int64_t m, int64_t n, double alpha, double *x, int64_t incx, double *y, int64_t incy, double *a, int64_t lda);
 
+/* LEVEL 1 BLAS FUNCTIONS */
+
+template <>
+void dot<double>(CUDABLAS_DOT_ARGTYPES(double)) {
+  cublasHandle_t handle = at::cuda::getCurrentCUDABlasHandle();
+  TORCH_CUDABLAS_CHECK(cublasDdot(handle, n, x, incx, y, incy, result));
+}
+
+template <>
+void dot<float>(CUDABLAS_DOT_ARGTYPES(float)) {
+  cublasHandle_t handle = at::cuda::getCurrentCUDABlasHandle();
+  TORCH_CUDABLAS_CHECK(cublasSdot(handle, n, x, incx, y, incy, result));
+}
+
+template <>
+void dot<at::Half>(CUDABLAS_DOT_ARGTYPES(at::Half)) {
+  cublasHandle_t handle = at::cuda::getCurrentCUDABlasHandle();
+  TORCH_CUDABLAS_CHECK(cublasDotEx(
```

Comment:
Hi,

https://github.com/pytorch/pytorch/blob/502ec8f7f76619e6cbb1efb0e23eb8c5438daed2/aten/src/THC/THCBlas.cu#L80-L94

ROCM case is missing for Half